mipengine/node/monetdb_interface/monet_db_connection.py

```diff
@@ -1,117 +1,183 @@
 from contextlib import contextmanager
-from time import sleep
+from functools import wraps
 from typing import List
 
 import pymonetdb
+from eventlet.greenthread import sleep
+from eventlet.lock import Semaphore
 
 from mipengine.node import config as node_config
 from mipengine.node import node_logger as logging
 from mipengine.singleton import Singleton
 
+INTEGRITY_ERROR_RETRY_INTERVAL = 1
 BROKEN_PIPE_MAX_ATTEMPTS = 50
-OCC_MAX_ATTEMPTS = 50
-INTEGRITY_ERROR_RETRY_INTERVAL = 0.5
+BROKEN_PIPE_ERROR_RETRY = 0.2
+CREATE_OR_REPLACE_QUERY_TIMEOUT = 1
+CREATE_REMOTE_TABLE_QUERY_TIMEOUT = 1
+INSERT_INTO_QUERY_TIMEOUT = (
+    node_config.celery.worker_concurrency * node_config.celery.run_udf_time_limit
+)
 
+create_remote_table_query_lock = Semaphore()
+create_function_query_lock = Semaphore()
+insert_query_lock = Semaphore()
 
-class MonetDB(metaclass=Singleton):
+
+class DBExecutionDTO:
+    def __init__(self, query, parameters=None, many=False):
+        self.query = query
+        self.parameters = parameters
+        self.many = many
+
+
+def query_execution_exception_handling(func):
```

Comment:
This is not just exception handling.

This also provides the `conn` object to the methods, which is very counter intuitive.

I can't think of a better way other than duplication of the handling in both methods, but at least add a better documentation and a better name to the wrapper.

Something like
`query_execution_conn_provider_and_exc_handling`
or something better...